.github/workflows/build_and_test_scala213_daily.yml

```diff
@@ -0,0 +1,115 @@
+name: Build and test Scala 2.13 daily
+
+on:
+  schedule:
+  - cron: '0 4 * * *'
+
+jobs:
+  # Build: build Spark and run the tests for specified modules.
+  build:
+    name: "${{ matrix.branch }}: ${{ matrix.modules }} ${{ matrix.comment }} (JDK ${{ matrix.java }}, ${{ matrix.hadoop }}, ${{ matrix.hive }})"
+    # Ubuntu 20.04 is the latest LTS. The next LTS is 22.04.
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        branch:
+          - master
+          - branch-3.2
+        java:
+          - 8
+        hadoop:
+          - hadoop3.2
+        hive:
+          - hive2.3
+        # TODO(SPARK-32246): We don't test 'streaming-kinesis-asl' for now.
+        # Kinesis tests depends on external Amazon kinesis service.
+        # Note that the modules below are from sparktestsupport/modules.py.
+        modules:
+          - >-
+            core, unsafe, kvstore, avro,
+            network-common, network-shuffle, repl, launcher,
+            examples, sketch, graphx
+          - >-
+            catalyst, hive-thriftserver
+          - >-
+            hive
+          - >-
+            sql
```

Comment:
This is a single `sql` module test which takes 1h 48min. Since this is a daily job, I avoid splitting the module.